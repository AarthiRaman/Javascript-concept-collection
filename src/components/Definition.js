import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const DefinitionContent = styled.p`
  margin: 10px 0 15px;
  text-align: left;
  font-weight: 500;
`;

function Definition({ title, textContent, notes }) {
  return (
    <div>
      <DefinitionContent>{textContent}</DefinitionContent>;
    </div>
  );
}

Definition.proptypes = {
  textContent: PropTypes.string
};

export default Definition;
