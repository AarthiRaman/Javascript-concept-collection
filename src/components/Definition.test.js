import React from "react";
import renderer from "react-test-renderer";

import Definition from "./Definition";

describe("Definition", () => {
  it("should render the component", () => {
    const comp = renderer
      .create(
        <Definition title="some title" textContent="some text">
          Facebook
        </Definition>
      )
      .toJSON();

    expect(comp).toMatchSnapshot();
  });
});
