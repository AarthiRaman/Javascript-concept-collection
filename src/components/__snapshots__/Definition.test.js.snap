<div>
  <h3
    className="sc-bdVaJa kxvujY"
  >
    some title
  </h3>
  ;
  <p
    className="sc-bwzfXH LQIkU"
  >
    some text
  </p>
  ;
</div>